app/forms/declaration/PackageInformation.scala

```diff
@@ -17,50 +17,86 @@
 package forms.declaration
 
 import forms.DeclarationPage
-import play.api.data.Forms.{number, text}
+import models.DeclarationType.{CLEARANCE, DeclarationType}
+import play.api.data.Forms.{number, optional, text}
 import play.api.data.{Form, Forms}
 import play.api.libs.json.{JsValue, Json}
 import services.PackageTypes
 import utils.validators.forms.FieldValidator._
 
-case class PackageInformation(typesOfPackages: String, numberOfPackages: Int, shippingMarks: String) {
+case class PackageInformation(typesOfPackages: Option[String], numberOfPackages: Option[Int], shippingMarks: Option[String]) {
+
+  def isEmpty: Boolean = typesOfPackages.isEmpty && numberOfPackages.isEmpty && shippingMarks.isEmpty
+
+  def nonEmpty: Boolean = !isEmpty
+
   def toJson: JsValue = Json.toJson(this)(PackageInformation.format)
 
-  def typesOfPackagesText: String = PackageTypes.findByCode(typesOfPackages).asText()
+  def typesOfPackagesText: Option[String] =
+    for {
```

Comment:
Nice.  But could you not just map the `typesOfPackages`?

Reply:
It could be
`typesOfPackages
    .flatMap(types => PackageTypes.findByCode(types)
    .map(_.asText()))`
But it is more readable with for comprehension in my opinion.